cli/src/cluster_query.rs

```diff
@@ -1747,8 +1744,58 @@ pub fn process_show_stakes(
 ) -> ProcessResult {
     use crate::stake::build_stake_state;
 
-    let progress_bar = new_spinner_progress_bar();
-    progress_bar.set_message("Fetching stake accounts...");
+    // fetching corresponding vote pubkey from identity pubkey
+    let vote_account_pubkeys = match vote_account_pubkeys {
+        Some(pubkeys) => {
+            let vote_account_progress_bar = new_spinner_progress_bar();
+            vote_account_progress_bar.set_message("Searching for matching vote accounts...");
+
+            let vote_accounts = rpc_client.get_vote_accounts()?;
+
+            let pubkeys: HashSet<String> =
+                pubkeys.iter().map(|pubkey| pubkey.to_string()).collect();
+
+            let filtered_vote_accounts: Vec<RpcVoteAccountInfo> = vote_accounts
+                .current
+                .iter()
+                .chain(vote_accounts.delinquent.iter())
+                .filter(|&vote_acc| {
+                    pubkeys.contains(&vote_acc.node_pubkey)
+                        || pubkeys.contains(&vote_acc.vote_pubkey)
+                })
+                .cloned()
+                .collect();
+
+            let all_pubkeys: HashSet<String> = filtered_vote_accounts
+                .iter()
+                .flat_map(|vote_acc| {
+                    vec![
+                        vote_acc.node_pubkey.to_string(),
+                        vote_acc.vote_pubkey.to_string(),
+                    ]
+                })
+                .collect();
+
+            let unmatched_pubkeys: HashSet<String> =
+                pubkeys.difference(&all_pubkeys).cloned().collect();
+
+            if !unmatched_pubkeys.is_empty() {
+                return Err(CliError::RpcRequestError(format!(
+                    "Failed to retrieve matching vote account for {:?}.",
+                    unmatched_pubkeys
+                ))
+                .into());
+            }
+            vote_account_progress_bar.finish_and_clear();
+
+            let filtered_vote_pubkeys = filtered_vote_accounts
+                .iter()
+                .map(|vote_acc| vote_acc.vote_pubkey.to_string())
+                .collect();
+            filtered_vote_pubkeys
```

Comment:
```suggestion
            let mut pubkeys: HashSet<String> =
                pubkeys.iter().map(|pubkey| pubkey.to_string()).collect();

            let vote_account_pubkeys: HashSet<String> = vote_accounts
                .current
                .into_iter()
                .chain(vote_accounts.delinquent)
                .filter_map(|vote_acc| {
                    (pubkeys.remove(&vote_acc.node_pubkey) || pubkeys.remove(&vote_acc.vote_pubkey))
                        .then(|| vote_acc.vote_pubkey)
                })
                .collect();

            if !pubkeys.is_empty() {
                return Err(CliError::RpcRequestError(format!(
                    "Failed to retrieve matching vote account for {:?}.",
                    pubkeys
                ))
                .into());
            }
            vote_account_progress_bar.finish_and_clear();
            vote_account_pubkeys
```
Would be nicer to iterate only once. Also, given all the Pubkey-String conversions, I prefer `clone()` to `to_string()` when needed to reallocate a String to make it more obvious what's occurring. The rest of this suggestion is to make the expression more concise.

Reply:
Nice suggestion! Very sorry I did not write it that way from the start, thanks.